lib/elixir/pages/Writing Documentation.md

```diff
@@ -48,6 +48,38 @@ The compiler will infer this argument as `map`. Sometimes the inference will be
 
     def size(map)
 
+## Functions with multiple clauses
+
+When documenting a function with multiple clauses, it is recommended to add the documentation
+and typespecs to the function head rather than adding @doc and @spec to each clause of
+the function.
```

Comment:
Also, I would rephrase this as:

> When documenting a function with multiple clauses, it is recommended to add a bodyless function head and add `@doc`s and `@spec`s to it.

This is also necessary when the function takes one or more optional arguments because you wouldn't know which clause to add the `\\` to :).